controllers/client/grafana_client.go

```diff
@@ -88,18 +57,26 @@ func NewGrafanaClient(ctx context.Context, c client.Client, grafana *v1beta1.Gra
 		return nil, errors.New("grafana admin secret does not contain password")
 	}
 
-	return &GrafanaClientImpl{
-		url:        grafana.Status.AdminUrl,
-		username:   username,
-		password:   password,
-		kubeClient: c,
-		httpClient: &http.Client{
+	userinfo := url.UserPassword(username, password)
+
+	clientConfig := grapi.Config{
+		APIKey:      "",
+		BasicAuth:   userinfo,
+		HTTPHeaders: nil,
+		Client: &http.Client{
 			Transport: transport,
 			Timeout:   time.Second * timeoutSeconds,
 		},
-	}, nil
-}
+		// TODO populate me
+		OrgID: 0,
+		// TODO populate me
+		NumRetries: 0,
```

Comment:
Just a comment:
Their library has a hardcoded value for delays between retries:
```go
// Wait a bit if that's not the first request
if n != 0 {
	time.Sleep(time.Second * 5)
}
```
([source](https://github.com/grafana/grafana-api-golang-client/blob/9c38b6f437bb9688f655ace9e66cc8b8f91ce0f4/client.go#L81))
5 sec is fine for terraform, because everything is taking ages there, but it's is too long for operators as they are expected to do many things in a timely manner. So, we'll need to either write our own wrapper that will implement retry logic or to try to prepare a PR to the upstream library that would make retry timeout more flexible.

Reply:
I think we can try to contribute this to the upstream library, I always prefer things to be working out-of-the-box, rather than having to do workarounds ourselves